### PR TITLE
web-wallet: Fix wrong error shown in the login screen

### LINKED ISSUE #3097: web-wallet: Misleading Error Message When Unlocking Wallet Without Internet Connection
### Description
Currently, when a user attempts to unlock their Wallet without an internet connection, they are presented with an **"invalid mnemonic" error**. This is misleading and may cause users to believe there is an issue with the mnemonic they entered, when the actual problem is the lack of internet connectivity.

As the Web Wallet uses a service worker, it is possible for the application to remain loaded in the browser even when there is no active internet connection. While this behavior is handled correctly in some areas (e.g., sync failures are displayed with appropriate status messages on the dashboard), it is not yet addressed in the pre-unlocked views.

**Affected Areas**
- Unlock screen
- Create wallet flow
- Restore wallet flow

### Expected Behavior
The application should properly handle scenarios where the user's internet connection is unavailable:
- Detect lack of internet connectivity.
- Display a clear and specific error message to inform the user about the connectivity issue.
- Prevent misleading error messages like "invalid mnemonic."

### Steps to Reproduce
- Load the Web Wallet while online.
- Disconnect from the internet.
- Attempt to unlock the wallet with a valid mnemonic.

### Actual Behavior
The user sees an **"invalid mnemonic" error**, despite entering a correct mnemonic.

### Proposed Solution
- Implement connectivity checks for the pre-unlocked views.
- Update error messaging to accurately reflect the connectivity status when applicable.

### Additional Context
This issue is particularly relevant as users may frequently encounter this scenario when working with service worker-based applications.

### LINKED ISSUE #3226: Web wallet failed sync - Invalid mnemonic phrase
**Description**
When the user tries to log into a web wallet when the web wallet can't sync with the node and gets an internal server error from the node, the wrong error message "Invalid mnemonic phrase" is shown to the user (even though the mnemonic did previously work).

**Steps To Reproduce**
When a user tries to log into a web wallet with the mnemonic (without a password), the node must return an internal server error to the web wallet, when the wallet tries to sync.

**Error Screenshots**
The internal server error popped up after the log in was successful but before the sync was performed:
image

After trying to log in again, a wrong "Invalid mnemonic phrase" error is shown:
image

If the dev console is open when trying to log in, the error below pops up. After continuing, the invalid mnemonic error pops up again, so they seem linked:
image

**Expected Behaviour**
The user should be able to log in without the node working, see the cached balance, but the "Failed sync" error message should be shown at the bottom of the main page:
image

If it's too difficult to let the user log in without the node, the error should say something like "Can't connect to blockchain":
image

**Platform**
Brave desktop browser (normal tab, not private)

### PR DESCRIPTION
**Pull request by ascartabelli**
Resolves #3226
Resolves #3097

### CODE REVIEW

**Comment by nortonandreev** [comment]
Was wondering if we should instead have this PR resolve this issue: https://github.com/dusk-network/rusk/issues/3097

Perhaps with the addition of an error if it cannot connect to the Network.

I don't think that the PR solves the core issues that have been outlined in https://github.com/dusk-network/rusk/issues/3226.

**Comment by ascartabelli** [comment]
I completely forgot we already had #3097 , I'll update the commit message tomorrow morning so that I include that as well.
Although I do think that this addresses one of the issues raised in #3226: the other ones are the same as #3227 and #3228 (which I consider as duplicates, and the core issue still needs to be investigated.

Another issue raised by the ones opened by @voidsculptorart , the one we talked about today, is the one about the proper loading of the protocol driver as we do have a bug about a case not handled with a graceful error message.
For this one I'll open a new issue.

**Comment by nortonandreev** [comment]
Screenshot 2024-12-19 at 22 44 57

Type error looks odd – but I guess it's not on our side. 🥴 Still better to what we had!

**Review by nortonandreev — APPROVED** [review]

**Review by nortonandreev — APPROVED** [review]

File: web-wallet/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fix wrong error shown in the login screen [#3226] [#3097]
```
Comment (nortonandreev) [inline web-wallet/CHANGELOG.md:20]: Bold!

Comment (ascartabelli) [reply web-wallet/CHANGELOG.md:20]: Very

**Pull request merged**